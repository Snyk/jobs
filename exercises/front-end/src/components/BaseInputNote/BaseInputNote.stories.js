import '~/assets/scss/styles.scss';
import BaseInputNote from './BaseInputNote';

export default {
  title: 'Library/Base Components/InputNote',
  component: BaseInputNote,
};

export const Default = () => ({
  components: { BaseInputNote },
  props: {
    variant: {
      default: 'info',
    },
  },
  template: `
    <BaseInputNote :variant="variant">
      Base Input Note ({{variant}}).
    </BaseInputNote>
  `,
});

export const Info = () => ({
  components: { BaseInputNote },
  template: `<BaseInputNote variant="info">A info input note for informational purposes.</BaseInputNote>`,
});

export const Warning = () => ({
  components: { BaseInputNote },
  template: `<BaseInputNote variant="warning">A warning input note for when there's a non-critical problem.</BaseInputNote>`,
});

export const Success = () => ({
  components: { BaseInputNote },
  template: `<BaseInputNote variant="success">A success input note for when something awesome has happened.</BaseInputNote>`,
});

export const Danger = () => ({
  components: { BaseInputNote },
  template: `<BaseInputNote variant="danger">A danger input note for when something unpleasant has happened.</BaseInputNote>`,
});
