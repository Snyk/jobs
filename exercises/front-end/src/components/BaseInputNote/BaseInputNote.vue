<template>
  <div class="vue--input-note" :class="computedClasses">
    <slot />
  </div>
</template>

<script>
import { isInList } from '~/lib/prop-validators';

/**
 * A note to appear under an input
 */
export default {
  name: 'BaseInputNote',

  status: 'under-review',

  props: {
    /**
     * Variant of BaseInputNote. Must be one of `default`, `info`, `warning`,
     * `error`, `danger`, `success`. Defaults to `default`
     */
    variant: {
      type: String,
      default: 'info',
      validator: isInList([
        'default',
        'info',
        'warning',
        'error',
        'danger',
        'success',
      ]),
    },
  },

  computed: {
    computedClasses() {
      return {
        [`vue--input-note--${this.variant}`]: true,
      };
    },
  },
};
</script>

<style lang="scss" scoped>
@import '~/assets/scss/utils';

.vue--input-note {
  @include typeset(caption);

  &--default,
  &--info {
    color: color(base, dimmed);
  }

  &--warning {
    color: color(alert, warning-text);
  }

  &--error,
  &--danger {
    color: color(alert, danger-text);
  }

  &--success {
    color: color(alert, success-text);
  }
}
</style>
