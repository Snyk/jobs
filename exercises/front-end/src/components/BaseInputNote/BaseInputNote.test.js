import { shallowMount } from '@vue/test-utils';
import BaseInputNote from '~/components/BaseInputNote/BaseInputNote';

describe('BaseInputNote', () => {
  const wrapper = shallowMount(BaseInputNote);

  it('renders as expected', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('Variants', () => {
    const wrapper = shallowMount(BaseInputNote);

    it('computes classes for error', async () => {
      await wrapper.setProps({
        variant: 'error',
      });

      expect(wrapper.vm.computedClasses).toEqual({
        'vue--input-note--error': true,
      });
    });

    it('computes classes for info', async () => {
      await wrapper.setProps({
        variant: 'info',
      });

      expect(wrapper.vm.computedClasses).toEqual({
        'vue--input-note--info': true,
      });
    });
  });
});
