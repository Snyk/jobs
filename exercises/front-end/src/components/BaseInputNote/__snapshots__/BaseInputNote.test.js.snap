// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseInputNote renders as expected 1`] = `<div class="vue--input-note vue--input-note--info"></div>`;
